genesyscloud/resource_genesyscloud_widget_deployment.go

```diff
@@ -0,0 +1,405 @@
+				Type:        schema.TypeBool,
+				Required:    true,
+			},
+			"disabled": {
+				Description: "When true, all create chat operations using this Deployment will be rejected.",
+				Type:        schema.TypeBool,
+				Required:    true,
+			},
+			"flow_id": {
+				Description: "The Inbound Chat Flow to run when new chats are initiated under this Deployment",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"allowed_domains": {
+				Description: "The list of domains that are approved to use this Deployment; the list will be added to CORS headers for ease of web use",
+				Type:        schema.TypeList,
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Required:    false,
+				Optional:    true,
+			},
+			"client_type": {
+				Description:  "The type of display widget for which this Deployment is configured, which controls the administrator settings shown.Valid values: v1, v2, v1-http, third-party.",
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringInSlice([]string{V1, V2, V1HTTP, THIRDPARTY}, false),
+			},
+
+			"client_config": {
+				Description: " The V1 and V1-http client configuration options that should be made available to the clients of this Deployment.",
+				Type:        schema.TypeSet,
+				MaxItems:    1,
+				Optional:    true,
+				Elem:        clientConfigSchemaResource,
+			},
+		},
+	}
+}
+
+func flattenClientConfig(clientType string, clientConfig platformclientv2.Widgetclientconfig) *schema.Set {
+	clientConfigSet := schema.NewSet(schema.HashResource(clientConfigSchemaResource), []interface{}{})
+
+	clientConfigMap := make(map[string]interface{})
+
+	if clientType == V1 {
+		if clientConfig.V1.WebChatSkin != nil {
+			clientConfigMap["webchat_skin"] = *clientConfig.V1.WebChatSkin
+		}
+
+		if clientConfig.V1.AuthenticationUrl != nil {
+			clientConfigMap["authentication_url"] = *clientConfig.V1.AuthenticationUrl
+		}
+	}
+
+	if clientType == V1HTTP {
+		if clientConfig.V1Http.WebChatSkin != nil {
+			clientConfigMap["webchat_skin"] = *clientConfig.V1Http.WebChatSkin
+		}
+
+		if clientConfig.V1Http.AuthenticationUrl != nil {
+			clientConfigMap["authentication_url"] = *clientConfig.V1Http.AuthenticationUrl
+		}
+	}
+
+	clientConfigSet.Add(clientConfigMap)
+
+	return clientConfigSet
+}
+
+func readWidgetDeployment(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sdkConfig := meta.(*providerMeta).ClientConfig
+	widgetsAPI := platformclientv2.NewWidgetsApiWithConfig(sdkConfig)
+
+	log.Printf("Reading widget deployment %s", d.Id())
+	return withRetriesForRead(ctx, 60*time.Second, d, func() *resource.RetryError {
+		currentWidget, resp, getErr := widgetsAPI.GetWidgetsDeployment(d.Id())
+
+		if getErr != nil {
+			if isStatus404(resp) {
+				return resource.RetryableError(fmt.Errorf("Failed to widget deployment %s: %s", d.Id(), getErr))
```

Comment:
Failed to widget deployment -> Failed to read widget deployment

Reply:
Fixed